library/src/cache/patch_manager.rs

```diff
@@ -229,6 +234,20 @@ impl PatchManager {
             }
         }
 
+        if let Some(public_key) = &self.patch_public_key {
+            // If we have a public key, verify that the patch has a signature
+            let patch_signature = patch
```

Comment:
It's really the hash that's signed rather than the whole patch, correct?

Reply:
Yep, updated to reflect this